### DISCUSSION TITLE
Styling edges with "(source/target)-endpoint property"

### DISCUSSION THREAD

**Discussion by shaharkazaz** [cytoscape/cytoscape.js#2961] [Q&A]
Hey 👋 

I'm not sure if this is a bug or intended behavior, hoping someone might have encountered this before.

I'm styling my edges with the following:
```json
{
      "width": 1,
      "line-color": "gray",
      "target-arrow-color": "gray",
      "target-arrow-shape": "triangle",
      "target-distance-from-node": 6,
      "source-distance-from-node": 6,
      "arrow-scale": 0.6,
      "source-endpoint": "outside-to-line-or-label",
      "target-endpoint": "outside-to-line-or-label",
      "curve-style": "bezier",
},
```

I noticed that the `distance-from-node` is respected when the edge is drawn from the node but ignored when drawn from the label:
![Screen Shot 2021-12-12 at 14 57 31](https://user-images.githubusercontent.com/17194830/145713316-2c8cb6c8-acc6-4631-aa96-0dd4e383a604.png)

I think it's related to the label's `text-margin-y` property which is set to `6`, when I set it to `0` the distance seems to work as expected but the label is positioned incorrectly.

Any idea how to resolve this? is this a bug? I would expect that the `text-margin-y` will be taken into consideratin.

**Comment by maxkfranz** [comment]
References:

- https://github.com/cytoscape/cytoscape.js/issues/1613
- https://github.com/cytoscape/cytoscape.js/issues/1599

It could be that a strict definition is being used w.r.t. node distance for one or more of those properties.  The references should give you some history and context to help diagnose your problem.
